### PR TITLE
TransactionProposalCreate edge cases and tests

### PR DESCRIPTION
**Pull request by Kassaking7**
### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

**Review by ckeshava — APPROVED** [review]

File: src/libxrpl/tx/transactors/proposal/TransactionProposalCreate.cpp [hunk 1]
```diff
@@ -61,8 +61,7 @@ TransactionProposalCreate::preflight(PreflightContext const& ctx)
         return temINVALID;
     }
 
-    if (proposedTx.isFieldPresent(sfFlags) &&
```
Comment (shawnxie999) [inline src/libxrpl/tx/transactors/proposal/TransactionProposalCreate.cpp]: I think we still need proposedTx.isFieldPresent(sfFlags) check, otherwise if the field doesn't exist it will throw. What if we move the STTx construction (line 128) to the start of preflight? would it avoid having need to check the existence of the fields

File: src/libxrpl/tx/transactors/proposal/TransactionProposalCreate.cpp [hunk 2]
```diff
@@ -104,6 +103,20 @@ TransactionProposalCreate::preflight(PreflightContext const& ctx)
     if (!proposedTx.isFieldPresent(sfTicketSequence) || proposedTx.getFieldU32(sfSequence) != 0)
         return temSEQ_AND_TICKET;
```
Comment (shawnxie999) [inline src/libxrpl/tx/transactors/proposal/TransactionProposalCreate.cpp:116]: based on recent discussion, we need to reserve ticket, so we should check if the ticket exists

**Review by shawnxie999 — APPROVED** [review]
